src/mono/mono/metadata/threads.c

```diff
@@ -3912,18 +3915,58 @@ mono_thread_init_apartment_state (void)
 		thread->apartment_state = ThreadApartmentState_Unknown;
 	}
 #endif
+
+#if TARGET_OSX || HOST_MACCAT || TARGET_IOS || TARGET_TVOS
+
+	ERROR_DECL (error);
+	MONO_STATIC_POINTER_INIT (MonoMethod, create_autoreleasepool)
+
+		create_autoreleasepool = mono_class_get_method_from_name_checked (mono_defaults.thread_class, "CreateAutoreleasePool", 0, 0, error);
+		mono_error_assert_ok (error);
+
+	MONO_STATIC_POINTER_INIT_END (MonoMethod, create_autoreleasepool)
+
+	if (mono_runtime_get_no_exec ())
+		return;
```

Comment:
I'm not 100% convinced the AOT compiler can get here, after all.  Let's drop it. The AOT compiler will break in CI if I'm wrong.

If it turns out this is necessary, I think it's slightly cleaner if the return is before the `create_autoreleasepool` initialization.

(Same thing for cleanup platform state).